### ISSUE TITLE
[BE] Add dependency injection for production code and tests

### ISSUE THREAD

**Issue by AnnaH3003** [amosproj/amos2023ws01-ticket-chat-ai#96]
## Tasks
- Replace direct instantiation of component dependencies with the Depends mechanism of FastAPI
- Refactor/Add unit tests to override these dependencies with mocks
- Refactor/Add integration tests with a mocked pymongo collection
- See [here for an example](https://github.com/amosproj/amos2023ws01-ticket-chat-ai/commit/249a92ce55a0eae25d051845539a2dcb431cd35e)

## Acceptance criteria
- [ ] All dependencies of components in the Backend are handled via Depends of FastAPI
- [ ] Mocking in the unit and integration tests is handled correctly

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
